dll/win32/ole32/CMakeLists.txt

```diff
@@ -58,31 +58,33 @@ list(APPEND SOURCE
     stg_stream.c
     storage32.c
     stubmanager.c
-    usrmarshal.c
-    precomp.h
-    ${CMAKE_CURRENT_BINARY_DIR}/dcom_i.c
-    ${CMAKE_CURRENT_BINARY_DIR}/irot_c.c)
+    usrmarshal.c)
 
 if(MSVC AND (ARCH STREQUAL "i386"))
     list(APPEND SOURCE msvc-thiscall.c)
 endif()
 
-add_library(ole32 MODULE
-    ${SOURCE}
+list(APPEND PCH_SKIP_SOURCE
     guid.c
-    ole32res.rc
     ${CMAKE_CURRENT_BINARY_DIR}/dcom_p.c
     ${CMAKE_CURRENT_BINARY_DIR}/ole32_objidl_p.c
     ${CMAKE_CURRENT_BINARY_DIR}/ole32_oleidl_p.c
     ${CMAKE_CURRENT_BINARY_DIR}/ole32_unknwn_p.c
     ${CMAKE_CURRENT_BINARY_DIR}/ole32_stubs.c
-    ${CMAKE_CURRENT_BINARY_DIR}/proxy.dlldata.c
+    ${CMAKE_CURRENT_BINARY_DIR}/proxy.dlldata.c)
+
+add_library(ole32 MODULE
+    ${SOURCE}
+    ${PCH_SKIP_SOURCE}
+    ole32res.rc
+    ${CMAKE_CURRENT_BINARY_DIR}/dcom_i.c
+    ${CMAKE_CURRENT_BINARY_DIR}/irot_c.c
```

Comment:
What does taking these out of SOURCE achieve? Previously that would mean they wouldn't get the PCH auto-included -- now being here just means the same as being in SOURCE, doesn't it?

Reply:
With the CMake built-in PCH functionality only exposing the ability to skip source files, I leveraged the existing framework to account for that, and seamlessly continue working in all the modules we're not skipping in. SOURCE was always meant to indicate source level files, as we put the autogenerated ones in the add_x() calls to make them visible.
Finally, SOURCE in the add_pch() calls that do not skip anything, are just that, the string "SOURCE" and so we're like "sure, I'll skip this source file named SOURCE from PCH when I see it" :)